src/lib/input/autosize.ts

```diff
@@ -72,9 +87,19 @@ export class MatTextareaAutosize implements AfterViewInit, DoCheck {
   ngAfterViewInit() {
     if (this._platform.isBrowser) {
       this.resizeToFitContent();
+
+      this._resize = this._ngZone.runOutsideAngular(() => {
+        return fromEvent(window, 'resize')
+          .pipe(debounceTime(10))
```

Comment:
wouldn't `auditTime` be more appropriate here?

Reply:
I don't think so. We want to prevent this from happening if we get a bunch of resize events close to each other.

Reply:
But in this case the rest of the page will be smoothly resizing and the textarea will just jump to the correct size at the end right? Wouldn't it be better to have it also smoothly resize?

Reply:
I think that the delay is so small that people won't really notice it if they're flipping their phone. The only ones that would would be developers, but I'd prioritize less repaints to dev satisfaction while resizing the window.

Reply:
the delay is small but if you get a resize event every 9ms for 1s, doesn't that mean the resize won't actually trigger until after 1s? (or I'm misunderstanding how `debounceTime` works)

"window of time required to wait for emission silence before emitting the most recent source value" (from http://reactivex.io/rxjs/class/es6/Observable.js~Observable.html#instance-method-debounceTime)

Reply:
It does, but the way this usually plays out is that the browser either fires a few of them at a time if you change your screen orientation, or one per pixel if you're resizing it manually. It's really hard to move your browser resize handle at a constant 1px per 9ms since it's about the speed of human reaction time.

Reply:
`auditTime` seems like a more natural fit for this though, since it will give us the most recent value every 10ms (or we could up it to 16ms which is ~60fps)

Reply:
Makes sense, I've switched it.